syntax = "proto3";

service TransferFileService {
	rpc UploadFile(stream UploadRequest) returns (UploadResponse) {}
	rpc DownloadFile(DownloadRequest) returns (stream DownloadResponse) {}
}

	message UploadRequest {
		bytes data = 1;
		int32 chunk_number = 2;
		string filename = 3;
		bool forwarded = 4;
	}

	message UploadResponse {
		bool success = 1;	
		string status_message = 2;
	}

	message DownloadRequest {
		string filename = 1;
	}

	message DownloadResponse {
		bytes data = 1;
		int32 chunk_number = 2;
	}
